### ISSUE TITLE
potential backports?

### ISSUE THREAD

**Issue by KingContaria** [RaphiMC/ImmediatelyFast#4]
Hi, are you considering making backports for older minecraft versions?

**Comment by RaphiMC** [comment]
I will not make backports of this mod, as the rendering engine changed/changes too often and I don't want to maintain different codebases for different minecraft versions

**Comment by Niwartilul** [comment]
I want to add a reminder to users of mods in general:
Yes, it sucks ass if a backport won't happen (or conversely a change or feature in general).
And asking nicely is not exactly wrong. But it's not all sunshine's and rainbows either.

I want you to consider something:
Most mod authors do what they do for free, rarely will they ever see a balancing income for the time spent, and even more rare is actually getting a profit. Typically income wise it's all negative. It's something people do as a hobby, either to sharpen their mind or to just take a idea and get it out of their head. OR to change something they are annoyed with, and out of their kindness (or rarely, for narcisstic purposes) share this work.

You have not purchased the mod, you are not dealing with a professional company nor the obligations of contracts...

Would you yourself be willing to spend the countless hours necessary to get something to work?
I don't mean you should insert some "Yhea I have considered the time it takes BUT" drivel in you asking for a backport. Just keep it in mind. 

In the particular case of backports. Mod/plugin developers are quite aware of the
possibility to backport. However is it worth it?

Infact, before you even ask for a backport, and with the time it takes in mind. Consider the possibility of doing the backport on your own. Most mods that don't deal with complicated math is something a regular joe can backport themselves with a few tweaks. In this particular mod, no. You probably won't be able to backport it on your own as a beginner. But in general, it's a good thing to try.
With the added benefit of learning something, which is always good.

And before you say "I don't have the time", consider it from the developers perspective too. If you do not have the time, how can you expect someone else to have it?

The only reason I went on this rant is because people are expecting shit handed to them on a silver platter without considering what it actually takes to do it. It's a thoughtless and quite selfish thing.
Many times I have considered asking the Fabric Discord for help, and 99% out of 100%, I managed to solve it on my own with sheer grit and perseverance. Learning to do things on your own as much as possible is a far better way to go. Yes this can mean weeks to solve it even if you are a retard like I am, but it's worth it in the end.

TL;DR think before you ask.

**Comment by StockiesLad** [comment]
Backported to 1.18.2 😄
